### ISSUE TITLE
实验复现结果不一致

### ISSUE THREAD

**Issue by AAHSY** [zyy-2001/DisKT#7]
请问下作者用的什么版本的torch和mamba_ssm

**Comment by zyy-2001** [comment]
> 请问下作者用的什么版本的torch和mamba_ssm
您好，感谢关注🙏🏻
1. 根据之前的实验，模型性能对torch版本敏感度不会太高，在可接受范围内即可。原来的版本使用的矛盾注意力是https://github.com/zyy-2001/DisKT/blob/89b742243cc03dbd33ebeaf724ba3b9a04edbd28/diskt/models/diskt.py#L260 这块，但被反馈这里可能有问题，背后的机理尚未确认。现在的版本更加严谨
2. 这个项目没有使用到mamba_ssm，DKT2用的xLSTM，抱歉当时没有详细记录版本号，但根据当时的时间轴来看大概是v1.0.5

如果有进一步的问题和发现，欢迎随时提出🌹

**Comment by AAHSY** [comment]
> > 请问下作者用的什么版本的torch和mamba_ssm
> > 您好，感谢关注🙏🏻
> 
> 1. 根据之前的实验，模型性能对torch版本敏感度不会太高，在可接受范围内即可。原来的版本使用的矛盾注意力是
>      
>        
>          [DisKT/diskt/models/diskt.py](https://github.com/zyy-2001/DisKT/blob/89b742243cc03dbd33ebeaf724ba3b9a04edbd28/diskt/models/diskt.py#L260)
>        
>        
>             Line 260
>          in
>          [89b7422](/zyy-2001/DisKT/commit/89b742243cc03dbd33ebeaf724ba3b9a04edbd28)
>        
>      
>      
>        
>    
>            
>              
>               # def contradictory_attention(query, key, value1, value2, mask=None, dropout=None, counter_attention_mask=None): 
>            
>        
>      
>    
>     这块，但被反馈这里可能有问题，背后的机理尚未确认。现在的版本更加严谨
> 2. 这个项目没有使用到mamba_ssm，DKT2用的xLSTM，抱歉当时没有详细记录版本号，但根据当时的时间轴来看大概是v1.0.5
> 
> 如果有进一步的问题和发现，欢迎随时提出🌹

感谢您的回复，我现在在同样的设备3090显卡上运行diskt模型，运行了CUDA_VISIBLE_DEVICES=0 python main.py --model_name diskt --data_name spanish这个命令，但是

<img width="599" height="222" alt="Image" src="https://github.com/user-attachments/assets/040b128c-616d-4530-bdea-636f11ca2caa" />只达到了这个效果，与论文相差较大

**Comment by zyy-2001** [comment]
> > > 请问下作者用的什么版本的torch和mamba_ssm
> > > 您好，感谢关注🙏🏻
> > 
> > 
> > 
> > 1. 根据之前的实验，模型性能对torch版本敏感度不会太高，在可接受范围内即可。原来的版本使用的矛盾注意力是
> >    ```
> >      [DisKT/diskt/models/diskt.py](https://github.com/zyy-2001/DisKT/blob/89b742243cc03dbd33ebeaf724ba3b9a04edbd28/diskt/models/diskt.py#L260)
> >    
> >    
> >         Line 260
> >      in
> >      [89b7422](/zyy-2001/DisKT/commit/89b742243cc03dbd33ebeaf724ba3b9a04edbd28)
> >    
> >    
> >    
> >    
> >    
> >        
> >          
> >           # def contradictory_attention(query, key, value1, value2, mask=None, dropout=None, counter_attention_mask=None): 
> >    ```
> >    
> >    
> >        
> >          
> >        
> >    
> >          
> >        
> >    
> >        
> >      
> >    这块，但被反馈这里可能有问题，背后的机理尚未确认。现在的版本更加严谨
> > 2. 这个项目没有使用到mamba_ssm，DKT2用的xLSTM，抱歉当时没有详细记录版本号，但根据当时的时间轴来看大概是v1.0.5
> > 
> > 如果有进一步的问题和发现，欢迎随时提出🌹
> 
> 感谢您的回复，我现在在同样的设备3090显卡上运行diskt模型，运行了CUDA_VISIBLE_DEVICES=0 python main.py --model_name diskt --data_name spanish这个命令，但是
> 
> <img alt="Image" width="599" height="222" src="https://private-user-images.githubusercontent.com/100771637/596685266-040b128c-616d-4530-bdea-636f11ca2caa.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Nzk0NTI5NzMsIm5iZiI6MTc3OTQ1MjY3MywicGF0aCI6Ii8xMDA3NzE2MzcvNTk2Njg1MjY2LTA0MGIxMjhjLTYxNmQtNDUzMC1iZGVhLTYzNmYxMWNhMmNhYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDUyMlQxMjI0MzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYjY0MDk2YjQ4ZmVmYWM4OTY0MjE2Y2YxZDRiN2JiZDBmYjk3MmFiZDRiZDA5MWFjZTg2MWQ1ZWE0ZTg0MDE1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.wXvB-qw1tbyoDT0gQ7rORLQzJI3rKkN2llrREOCDWa4">只达到了这个效果，与论文相差较大

您可以把最后那个矛盾注意力取消注释看看~

**Comment by AAHSY** [comment]
感谢您的回复！我取消矛盾注意力的注释之后，效果变化微乎其微呢

**Comment by zyy-2001** [comment]
> 感谢您的回复！我取消矛盾注意力的注释之后，效果变化微乎其微呢

您好，我刚下载测试了一下，这是取消注释后的结果：

<img width="862" height="129" alt="Image" src="https://github.com/user-attachments/assets/6d408b8f-357c-4248-b3c4-482b5d19aeff" />

我现在的torch版本是2.9.0，我不清楚这是否是版本带来的性能原因，但我印象中torch带来的性能差异应该在0.05内，我当时使用的应该是CUDA 11.8，实验应该在前年年底左右进行的，使用的是当时CUDA 11.8下torch的最新版。您可以多尝试其他不同版本的torch观察一下性能的敏感程度，不过不建议在spanish这个数据集下进行，这个数据集太小了，性能差异可能比较大。抱歉很久没做该任务了~

**Comment by AAHSY** [comment]
可以告诉我您现在环境里一些包的版本吗，我已经配了三次环境效果都不理想，麻烦您

了

**Comment by zyy-2001** [comment]
> > 感谢您的回复！我取消矛盾注意力的注释之后，效果变化微乎其微呢
> 
> 您好，我刚下载测试了一下，这是取消注释后的结果：
> 
> <img alt="Image" width="862" height="129" src="https://private-user-images.githubusercontent.com/73881472/597104789-6d408b8f-357c-4248-b3c4-482b5d19aeff.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Nzk1MjI0MjcsIm5iZiI6MTc3OTUyMjEyNywicGF0aCI6Ii83Mzg4MTQ3Mi81OTcxMDQ3ODktNmQ0MDhiOGYtMzU3Yy00MjQ4LWIzYzQtNDgyYjVkMTlhZWZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNTIzVDA3NDIwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjMDYwOTYyN2Y1ZTA2YzFlY2NlNWUyZWQzMWZiZTQ0M2YzZjUyMjc4MTMxMDI4ZDc2OWI1ZDNmOWFkZTQ3ZjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.cuR1EAc_ZKYH0ecG1tDI_RSd4u6ba-RIw69p6MzGPKE">
> 我现在的torch版本是2.9.0，我不清楚这是否是版本带来的性能原因，但我印象中torch带来的性能差异应该在0.05内，我当时使用的应该是CUDA 11.8，实验应该在前年年底左右进行的，使用的是当时CUDA 11.8下torch的最新版。您可以多尝试其他不同版本的torch观察一下性能的敏感程度，不过不建议在spanish这个数据集下进行，这个数据集太小了，性能差异可能比较大。抱歉很久没做该任务了~

我刚更新了，您可以找下当时实验下的配置，torch的临近版本带来的性能差异应该不会太大

**Comment by AAHSY** [comment]
好的，我去试试，感谢您及时回复！
